File: frontend/src/services/orderService.js
import apiClient from './apiClient';
import { ENDPOINTS } from '../constants/apiEndpoints';

export const orderService = {
  createOrder: async (orderData) => {
    const response = await apiClient.post(ENDPOINTS.ORDERS, orderData);
    return response.data.data;
  },

  getUserOrders: async () => {
    const response = await apiClient.get(ENDPOINTS.ORDERS);
    return response.data.data;
  },

  getOrderByNumber: async (orderNumber) => {
    const response = await apiClient.get(`${ENDPOINTS.ORDERS}/${orderNumber}`);
    return response.data.data;
  },
};

File: backend/src/main/java/com/yourorg/appname/service/OrderService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.request.OrderRequest;
import com.yourorg.appname.dto.response.OrderResponse;

import java.util.List;

public interface OrderService {
    OrderResponse createOrder(OrderRequest request, String username);
    List<OrderResponse> getUserOrders(String username);
    OrderResponse getOrderByNumber(String orderNumber, String username);
}
